### PR TITLE
add boundary variables for the soil model

### PR DESCRIPTION
**Pull request by kmdeck**
## Purpose 
Adds the ability to add boundary condition specific variables to the cache/aux state for the soil model. This is not required for other standalone models currently. This 
1. is required for runoff, since that requires additional global parameter fields (assuming these will be stored in aux!)
2.  adds clarity in atmos-driven soil runs regarding where soil SHF, LHF, E, Rn get stored [now in `p.soil` instead of `p`]

## Content
Source files changed: `soil_canopy_model.jl`, `Soil/boundary_conditions.jl`, `Soil/rre.jl`, `Soil/energy_hydrology.jl`
Tests added
Changes propagated to experiments and docs

Source code changes:
1. Add `boundary_var` types, names, and domains functions (soil standalone files). This follows the style we use for setting aux variables for any model, but dispatches off of the type of BC set for the soil model. The default is to have only the top_bc and bottom_bc as variables.
2. These aux vars are updated prior in `update_boundary_fluxes!`
3. Removes the "soil_SHF", "soil_LHF" etc. fields from the interaction variables of the integrated model, since these can now be stored in `p.soil` instead of in `p` directly.

Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by kmdeck** [comment]
Aqua.test_project_toml_formatting gave an undefined error. Maybe it doesnt exist anymore? 
https://juliatesting.github.io/Aqua.jl/stable/#test_functions

File: test/aqua.jl [hunk 1]
```diff
@@ -27,8 +27,7 @@ end
     Aqua.test_stale_deps(ClimaLSM)
     Aqua.test_deps_compat(ClimaLSM)
     Aqua.test_project_extras(ClimaLSM)
-    Aqua.test_project_toml_formatting(ClimaLSM)
-    Aqua.test_piracy(ClimaLSM)
+    Aqua.test_piracies(ClimaLSM)
```
Comment (juliasloan25) [inline test/aqua.jl:30]: thank you for updating this!!

could you also add a compat entry `Aqua = "0.8"` in test/Project.toml please?

File: src/standalone/Soil/boundary_conditions.jl [hunk 2]
```diff
@@ -127,6 +216,49 @@ function AtmosDrivenFluxBC(atmos, radiation; runoff = NoRunoff())
     return AtmosDrivenFluxBC{typeof.(args)...}(args...)
 end
 
+"""
+    boundary_vars(::AtmosDrivenFluxBC)
```
Comment (juliasloan25) [inline src/standalone/Soil/boundary_conditions.jl:220]: docstring doesn't match function

File: src/standalone/Soil/boundary_conditions.jl [hunk 3]
```diff
@@ -127,6 +216,49 @@ function AtmosDrivenFluxBC(atmos, radiation; runoff = NoRunoff())
...
+adds the surface conditions (SHF, LHF, evaporation, and resistance) and the
+net radiation to the auxiliary variables.
+
+These variables are updated in place in `soil_boundary_fluxes`.
+"""
+boundary_vars(bc::AtmosDrivenFluxBC, ::ClimaLSM.TopBoundary) =
+    (:sfc_conditions, :R_n)
+
+"""
+    boundary_var_domain_names(::AtmosDrivenFluxBC)
```
Comment (juliasloan25) [inline src/standalone/Soil/boundary_conditions.jl:232]: docstring

File: test/standalone/Soil/climate_drivers.jl [hunk 4]
```diff
@@ -145,8 +148,8 @@ for FT in (Float32, Float64)
             init_soil!(Y, coords.subsurface.z, model.parameters)
             set_initial_aux_state! = make_set_initial_aux_state(model)
             set_initial_aux_state!(p, Y, t)
-
-
+            tendency! = make_exp_tendency(model)
+            tendency!(similar(Y), Y, p, t)
```
Comment (juliasloan25) [inline test/standalone/Soil/climate_drivers.jl:152]: why do we need to call `tendency!` here now? I would expect `p` to get set with the initial `Y` when we call `set_initial_aux_state!(p, Y, t)`, so then calling `tendency!(...)` and e.g. `R_n = ClimaLSM.net_radiation(...)` would make `R_n` one timestep ahead of `p.soil.R_n`

File: test/standalone/Soil/soil_bc.jl [hunk 5]
```diff
@@ -198,3 +198,13 @@ for FT in (Float32, Float64)
         @test abs(flux_expected - flux_int) < eps(FT)
     end
 end
+
+
+@testset "Boundary vars" begin
+    # AtmosDrivenFluxBC method tested in `test/standalone/Soil/climate_drivers.jl`
+    # Currently no other methods exist
+    struct Foo <: Soil.AbstractSoilBC end
+    @test Soil.boundary_vars(Foo(), ClimaLSM.TopBoundary()) == ()
```
Comment (juliasloan25) [inline test/standalone/Soil/soil_bc.jl:207]: should we test with `ClimaLSM.BottomBoundary()` as the second argument as well?

File: src/integrated/soil_canopy_model.jl [hunk 6]
```diff
@@ -294,10 +277,7 @@ function make_interactions_update_aux(
 
         # Soil boundary fluxes under canopy or for bare soil
         bc = land.soil.boundary_conditions.top
-        soil_conditions = surface_fluxes(bc.atmos, land.soil, Y, p, t)
-        @. p.soil_shf = soil_conditions.shf
-        @. p.soil_evap = soil_conditions.vapor_flux
-        @. p.soil_lhf = soil_conditions.lhf
+        p.soil.sfc_conditions .= surface_fluxes(bc.atmos, land.soil, Y, p, t)
```
Comment (juliasloan25) [inline src/integrated/soil_canopy_model.jl:280]: nice change!

**Review by juliasloan25 — COMMENTED** [review]
Looks good! I had a few questions

**Comment by juliasloan25** [comment]
> Aqua.test_project_toml_formatting gave an undefined error. Maybe it doesnt exist anymore?
> https://juliatesting.github.io/Aqua.jl/stable/#test_functions

It was removed in Aqua v0.8.0 (see release notes [here](https://github.com/JuliaTesting/Aqua.jl/releases)). We should have had a compat entry for `0.7` before

File: test/standalone/Soil/soil_bc.jl (hunk 5 above)
Comment (kmdeck) [reply test/standalone/Soil/soil_bc.jl:207]: I added it in!

File: test/standalone/Soil/climate_drivers.jl (hunk 4 above)
Comment (kmdeck) [reply test/standalone/Soil/climate_drivers.jl:152]: yes, there is something not great going on! We arent actually stepping the state `Y` when we call tendency!, just computing `dY`, and the `dY` is used to increment `Y` in a separate step. so `p` and `Y` should still be in sync after calling `tendency!`. I think the issue is that we have some aux variables that are updated in `update_aux!` and some that are updated in the tendency. 

it would be great if we had a way around this.  for standalone models I think we could move things so that the boundary fluxes are computed in `update_aux!`, and then just used in the tendency function. (The boundary fluxes depending on R_n, SHF, LHF, etc). But with integrated models, things get more complex, because canopy aux variables can affect the boundary conditions for the soil, and vice versa. it's worth more thought though. because `update_aux!` is becoming more misleading as a name. Also, the forcing variables are in the cache too, but will get update first in a separate function (at least in the idea we had discussed earlier).

Comment (kmdeck) [reply test/standalone/Soil/climate_drivers.jl:152]: I vaguely wonder about:
```
function update_aux!(p,Y,t)
    update_forcing!(p,Y,t)
    update_standalone_aux!(p,Y,t) # our current update aux
    update_boundary_conditions!(p,Y,t)
end
```
or something? TBD 😵‍💫

File: experiments/standalone/Soil/evaporation.jl [hunk 7]
```diff
@@ -262,14 +271,13 @@ for (FT, tf) in ((Float32, 2 * dt), (Float64, tf))
...
             push!(evap_0, vapor_flux)
-            push!(L_MO, potential_conditions.L_MO)
         end
 
         plt1 = Plots.plot()
         Plots.plot!(
             plt1,
             sol.t ./ 3600 ./ 24,
-            (evap .* r_ae ./ (r_ae .+ r_soil)) ./ evap_0,
+            evap ./ evap_0,
```
Comment (kmdeck) [inline experiments/standalone/Soil/evaporation.jl:280]: evap, computed by `surface_conditions`, has already accounted for r_soil.

File: test/standalone/Soil/climate_drivers.jl (hunk 4 above)
Comment (juliasloan25) [reply test/standalone/Soil/climate_drivers.jl:152]: we could move towards atmos's naming style and use `update_cache_vars!` instead of `update_aux!`? though maybe it would be confusing if our "cache" means something slightly different from theirs

File: test/integrated/lsms.jl [hunk 8]
```diff
@@ -151,9 +151,11 @@ for FT in (Float32, Float64)
             m::DummyModel3{FT},
```
Comment (juliasloan25) [inline test/integrated/lsms.jl:151]: I can't add a comment on those lines but I don't think we need `make_update_aux` for DummyModel4 or DummyModel1

File: test/integrated/lsms.jl [hunk 9]
```diff
@@ -151,9 +151,11 @@ for FT in (Float32, Float64)
...
         ) where {FT}
             update_aux! = ClimaLSM.make_update_aux(m)
+            update_boundary_fluxes! = ClimaLSM.make_update_boundary_fluxes(m)
```
Comment (juliasloan25) [inline test/integrated/lsms.jl:154]: should `update_aux` for `DummyModel4` also call `update_boundary_fluxes`?

File: test/standalone/Soil/climate_drivers.jl [hunk 10]
```diff
@@ -119,6 +119,11 @@ for FT in (Float32, Float64)
             )
 
             Y, p, coords = initialize(model)
+            @test propertynames(p.soil.sfc_conditions) ==
+                  (:lhf, :shf, :vapor_flux, :r_ae)
+            @test :top_bc ∈ propertynames(p.soil)
```
Comment (juliasloan25) [inline test/standalone/Soil/climate_drivers.jl:124]: could we instead test
```
@test propertynames(p.soil.sfc_conditions) ==
       (:sfc_conditions, :top_bc, :bottom_bc, :R_n)
```
?
Maybe this would be a stricter test bc it would also check that we don't add anything extra

File: docs/tutorials/standalone/Usage/model_tutorial.jl [hunk 11]
```diff
@@ -0,0 +1,369 @@
+# The `AbstractModel` framework allows users to define land
```
Comment (juliasloan25) [inline docs/tutorials/standalone/Usage/model_tutorial.jl:1]: great tutorial! :)

File: test/integrated/pond_soil_lsm.jl [hunk 12]
```diff
@@ -113,25 +113,23 @@ for FT in (Float32, Float64)
         land_auxiliary_vars = auxiliary_vars(land)
         land_auxiliary_types = auxiliary_types(land)
         # prognostic vars
-        @test land_prognostic_vars.soil == prognostic_vars(land.soil)
-        @test land_prognostic_vars.surface_water ==
+        @test land_prognostic_vars[1].soil == prognostic_vars(land.soil)
```
Comment (juliasloan25) [inline test/integrated/pond_soil_lsm.jl:116]: why do we index into the vars with integer indices? I don't understand the structure of prognostic and aux vars here

File: docs/tutorials/standalone/Usage/LSM_single_column_tutorial.jl [hunk 13]
```diff
@@ -298,26 +271,34 @@ land = LandHydrology{FT}(;
...
-# or cache state: storing a quantity which we would rather compute once
-# and store, rather than compute twice, once in the soil tendency function,
-# and once in the pond tendency function. This guarantees the same value is
-# used for both equations. In pseudo code, we have:
+# in `p.soil_infiltration`.  In pseudo code, we have:
 # ```julia
 # function make_update_aux(land)
 #          soil_update_aux! = make_update_aux(land.soil)
 #          surface_update_aux! = make_update_aux(land.surface_water)
-#          interactions_update_aux! = make_update_aux(land, land.soil, land.surface_water)
```
Comment (juliasloan25) [inline docs/tutorials/standalone/Usage/LSM_single_column_tutorial.jl:310]: do we need to replace this with `initialize_additional_lsm_aux`?

File: experiments/integrated/ozark/ozark.jl [hunk 14]
```diff
@@ -356,6 +335,9 @@ function diurnal_avg(series)
 end
 
 # Heterotrophic respiration
+# Calculate RH as boundary flux
```
Comment (juliasloan25) [inline experiments/integrated/ozark/ozark.jl:338]: typo: RH -> HR

Comment (juliasloan25) [reply experiments/integrated/ozark/ozark.jl:338]: also we aren't really calculating HR anymore, just accessing it

File: experiments/integrated/vaira_ranch/vaira.jl [hunk 15]
```diff
@@ -356,7 +335,9 @@ function diurnal_avg(series)
 end
 
 # Heterotrophic respiration
-
+# Calculate RH as boundary flux
```
Comment (juliasloan25) [inline experiments/integrated/vaira_ranch/vaira.jl:338]: typo: RH -> HR

File: experiments/standalone/Soil/water_conservation.jl [hunk 16]
```diff
@@ -232,32 +232,28 @@ for FT in (Float32, Float64)
...
-                TopBoundary(),
-                soil_dirichlet,
-                Δz_top,
-                sol.u[j],
-                p,
-                times[j],
-            )
-            flux_in_sim[j - 1] = parent(top_flux_bc)[1]
-        end
+        flux_in_sim =
```
Comment (juliasloan25) [inline experiments/standalone/Soil/water_conservation.jl:254]: nice change!

**Review by juliasloan25 — COMMENTED** [review]
I had a lot of comments but this looks really good!! Thank you for all the changes :)

File: src/ClimaLSM.jl [hunk 17]
```diff
@@ -153,17 +156,20 @@ function make_exp_tendency(land::AbstractLandModel)
...
         update_aux!(p, Y, t)
+        update_boundary_fluxes!(p, Y, t)
         for f! in compute_exp_tendency_list
             f!(dY, Y, p, t)
         end
     end
     return exp_tendency!
 end
 
+# this needs thought
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/ClimaLSM.jl:170]: I think we are OK because when the user calls `initialize(model)`, `initialize_additional_lsm_aux` will get called and so those variables will appear in `p`, but with zero as value. Then in `make_set_initial_aux_state(model)`, we call and use ` update_boundary_fluxes! = make_update_boundary_fluxes(land)` , which will set the values correctly.

File: src/ClimaLSM.jl [hunk 18]
```diff
@@ -222,65 +221,60 @@ function prognostic_vars(land::AbstractLandModel)
...
         auxiliary_vars(getproperty(land, model))
     end
-    return NamedTuple{(components..., :interactions)}((
-        auxiliary_list...,
-        interaction_vars(land),
-    ))
+    return (add_lsm_aux_vars(land)..., NamedTuple{components}(auxiliary_list))
 end
 
 function auxiliary_types(land::AbstractLandModel)
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/ClimaLSM.jl:244]: add back in as part of PR#417

File: src/ClimaLSM.jl [hunk 19]
```diff
@@ -105,26 +105,27 @@ function initialize_auxiliary(
 end
 
 """
-    initialize_interactions(land::AbstractLandModel) end
+    initialize_additional_lsm_aux(land::AbstractLandModel) end
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/ClimaLSM.jl:108]: I shortened to "initialize_add_lsm_aux" in PR#417. this is consistent with the naming where we define the variables, `add_lsm_aux_vars`, for example.

File: src/integrated/pond_soil_model.jl [hunk 20]
```diff
@@ -253,10 +259,11 @@ stored in `p.soil_infiltration`).
 function ClimaLSM.boundary_flux(
     bc::RunoffBC,
     ::TopBoundary,
+    model::Soil.RichardsModel,
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/integrated/pond_soil_model.jl:262]: The signature for `boundary_flux` for soil models uses the `model`, e.g.:https://github.com/CliMA/ClimaLSM.jl/blob/f667b01789714e77cc3ad91e1fe7c3662ef67f44/src/standalone/Soil/rre.jl#L103

I think not passing `model` was a bug, but since this is mainly an example in a tutorial explaining how to set up LSMs, im not sure we ever really run a simulation and check the output. so i think this was a case of insufficient testing.

[the remark this answers is not in the document]
Comment (kmdeck) [reply src/integrated/pond_soil_model.jl:262]: will update docs & add type enforcement in PR417

File: test/integrated/lsms.jl (hunk 8 above)
Comment (kmdeck) [reply test/integrated/lsms.jl:151]: How come?

File: test/integrated/lsms.jl (hunk 9 above)
Comment (kmdeck) [reply test/integrated/lsms.jl:154]: Right now, `update_aux` is a separate function than `update_boundary_fluxes`. They both update parts of the cache, but not the same variables.

File: docs/tutorials/standalone/Usage/model_tutorial.jl (hunk 11 above)
Comment (kmdeck) [reply docs/tutorials/standalone/Usage/model_tutorial.jl:1]: will add in as part of #417 !

File: docs/tutorials/standalone/Usage/LSM_single_column_tutorial.jl (hunk 13 above)
Comment (kmdeck) [reply docs/tutorials/standalone/Usage/LSM_single_column_tutorial.jl:310]: fixed in PR#417!

File: test/integrated/pond_soil_lsm.jl (hunk 12 above)
Comment (kmdeck) [reply test/integrated/pond_soil_lsm.jl:116]: yeah, Im not sure what to do here. Before, we returned a NamedTuple, for prognostic and aux vars, so we could access them like: `land_prognostic_vars.soil`, without the indexing. This is slightly inconsistent between integrated and standalone models, since `prognostic_vars(standalone model)` returns a tuple (e.g. `prognostic_vars(::RichardsModel) = (:\vartheta_l,)`. But for integrated models, we would make use of the names and return `(:soil = (:\vartheta_l,), :canopy = (...))`.

The additional aux variables were stored in the NamedTuple with the key `interactions`. I wasnt sure that was clear anymore, but without a key, returning a Tuple felt like it made the most sense. And if I did it for aux, I felt like it was good to be consistent with prognostic. but then we get the indexing.

Maybe I can change back to how it was (NamedTuples), and just replace the key `:interactions` with `lsm_aux` ?

Comment (kmdeck) [reply test/integrated/pond_soil_lsm.jl:116]: I think I will do that! it will result in a smaller change in the PR compared with main, and we can do a one off PR later to address it if we feel like we need to. when we add forcing variables, it may change a little also.

File: src/ClimaLSM.jl (hunk 19 above)
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/ClimaLSM.jl:108]: My main worry would be that if it was `initialize_lsm_aux` alone, someone might think this was all of the aux vars for the LSM?

[the remark this answers is not in the document]
Comment (kmdeck) [reply src/ClimaLSM.jl:108]: but, the doc string is clarifying that this adds additional variables. maybe that is the nicer name, then.

[the remark this answers is not in the document]
Comment (kmdeck) [reply src/ClimaLSM.jl:108]: lol, ok! im going for `initialize_lsm_aux` :P and this is consistent also with the NamedTuple fix below

File: src/integrated/soil_canopy_model.jl [hunk 21]
```diff
@@ -383,7 +366,8 @@ function lsm_radiant_energy_fluxes!(
 
 
     # net soil = (1-α)*trans for par and nir
-    @. SW_net_soil =
+    R_net_soil .= FT(0)
+    @. R_net_soil +=
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/integrated/soil_canopy_model.jl:370]: I changed this.

File: src/standalone/Soil/boundary_conditions.jl [hunk 22]
```diff
@@ -13,6 +16,129 @@ An abstract type for soil-specific types of boundary conditions, like free drain
...
+
+Use this function in the exact same way you would use `auxiliary_types`.
+"""
+boundary_var_types(
+    ::Soil.RichardsModel{FT},
+    ::NamedTuple,
+    ::ClimaLSM.AbstractBoundary,
+) where {FT} = (FT,)
+
+#=
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Soil/boundary_conditions.jl:88]: we actually dont need these - will remove!

File: src/standalone/Soil/rre.jl [hunk 23]
```diff
@@ -249,16 +247,53 @@ ClimaLSM.prognostic_domain_names(soil::RichardsModel) = (:subsurface,)
...
-We could instead compute the conductivity and matric potential within the
-tendency function explicitly, rather than compute and store them in the
-auxiliary vector `p`. We did so in this case as a demonstration.
 """
-ClimaLSM.auxiliary_vars(soil::RichardsModel) = (:K, :ψ, :top_bc, :bottom_bc)
-ClimaLSM.auxiliary_types(soil::RichardsModel{FT}) where {FT} = (FT, FT, FT, FT)
-ClimaLSM.auxiliary_domain_names(soil::RichardsModel) =
-    (:subsurface, :subsurface, :surface, :surface)
+"""
+function auxiliary_domain_names(soil::RichardsModel)
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Soil/rre.jl:265]: will do!

File: test/standalone/Soil/climate_drivers.jl (hunk 10 above)
Comment (kmdeck) [reply test/standalone/Soil/climate_drivers.jl:124]: updated this!

File: src/standalone/Vegetation/Canopy.jl [hunk 24]
```diff
@@ -556,7 +559,7 @@ function ClimaLSM.make_update_aux(
             GPP,
             Rd,
             β,
-            h,
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Vegetation/Canopy.jl]: previously we were passing it h (from the atmos struct), but this is incorrect. we should be passing in h_canopy.

File: src/standalone/Soil/boundary_conditions.jl [hunk 25]
```diff
@@ -13,6 +16,129 @@ An abstract type for soil-specific types of boundary conditions, like free drain
...
+top and bottom boundary flux fields,
+but which can be extended depending on the type of boundary condition used.
+
+Use this function in the exact same way you would use `auxiliary_types`.
+"""
+boundary_var_types(
+    ::Soil.RichardsModel{FT},
+    ::NamedTuple,
+    ::ClimaLSM.AbstractBoundary,
+) where {FT} = (FT,)
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Soil/boundary_conditions.jl:86]: potentially? I could go either way on this. Right now these are the two use cases. I think if we made this NamedTuple typed, it would just be for consistency. We would use it like:
```
    p.soil.top_bc.water .= boundary_flux(
            model.boundary_conditions.top.water,
            TopBoundary(),
            model,
            Δz_top,
            Y,
            p,
            t,
        )
```

instead of 
```
    p.soil.top_bc .= boundary_flux(
            model.boundary_conditions.top.water,
            TopBoundary(),
            model,
            Δz_top,
            Y,
            p,
            t,
        )
```

but I think there would be little else that changed. what is your preference?

[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Soil/boundary_conditions.jl:86]: for now, I added to the doc string explaining why this is a scalar while the EnergyHydrology method returns a NamedTuple.

File: src/standalone/Soil/boundary_conditions.jl [hunk 26]
```diff
@@ -187,21 +360,21 @@ function soil_boundary_fluxes(
     t,
 ) where {FT}
 
-    conditions = surface_fluxes(bc.atmos, model, Y, p, t)
-    R_n = net_radiation(bc.radiation, model, Y, p, t)
+    p.soil.sfc_conditions .= surface_fluxes(bc.atmos, model, Y, p, t)
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Soil/boundary_conditions.jl:313]: Ill follow up with another PR that changes the naming to `turbulent_fluxes` and also the name of the function we call in line 313 to match.

File: src/standalone/Soil/boundary_conditions.jl (hunk 25 above)
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Soil/boundary_conditions.jl:86]: I think youre right! we also pass in the BC as NamedTuples, so we should do that here. ill make this change.

File: experiments/integrated/fluxnet/data_tools.jl [hunk 27]
```diff
@@ -93,35 +72,35 @@ function VerifiedColumn(file_matrix::Matrix, column_name::String, units::String)
...
-    if QCstatus != absent && minimum(QC_col) != 2
-        # Fill any missing data with the mean value per the QC flag
-        num_missing = count(x -> x > 1, QC_col)
-        if num_missing > 0
-            perc_missing = 100.0 * num_missing / length(col_dat)
-            @info "Warning: Data for $column_name $(
-            round(perc_missing, sigdigits=3))% missing. Filling with mean value"
-            replace_missing_with_mean!(col_dat, QC_col)
+    elseif status == complete
+        return DataColumn(col_dat, units, status)
```
Comment (Espeer5) [inline experiments/integrated/fluxnet/data_tools.jl:76]: One thing here is that we need to check for the QC flag and fill using the QC flag before we can consider the column complete. `check_column` only checks for missings by value (`==-9999`), but in the datasets there are values that are missing or very low quality according to the QC flag but actually have data populated in the normal range. As a result, `check_column` can return status complete when the QC flag will say something different.

Alternatively we could add checking QC to the `check_column` function to neaten the logic here

Comment (Espeer5) [reply experiments/integrated/fluxnet/data_tools.jl:76]: My original goal here was that `check_column` just ensured that column existed in the file, and then `filter_column` processed it for missing values which is why I didn't include the QC in `check_column`

Comment (kmdeck) [reply experiments/integrated/fluxnet/data_tools.jl:76]: ah ok! Ill try and adjust this to reflect that, and then re-request review

File: experiments/integrated/fluxnet/data_tools.jl [hunk 28]
```diff
@@ -63,28 +63,7 @@ struct DataColumn
...
-        @info "Warning: Data for $column_name $(round(percent_missing,
-        sigdigits=3))% missing. Filling with mean value"
-        replace_missing_with_mean_by_value!(col_dat)
-        status = complete
-    end
-    return DataColumn(col_dat, units, status)
-end
-
-function VerifiedColumn(file_matrix::Matrix, column_name::String, units::String)
+function filter_column(file_matrix::Matrix, column_name::String, units::String)
```
Comment (juliasloan25) [inline experiments/integrated/fluxnet/data_tools.jl:66]: Are these changes related to the boundary vars change? It might be nice to have them in a separate PR if not, but I don't think it's a huge deal either way

File: test/integrated/lsms.jl (hunk 8 above)
Comment (juliasloan25) [reply test/integrated/lsms.jl]: Ahh I was thinking they don't get called, but actually I think I was wrong. It'll get called for both of them in the last testset `make_set_initial_aux_state(m)`

File: src/ClimaLSM.jl (hunk 19 above)
[the remark this answers is not in the document]
Comment (juliasloan25) [reply src/ClimaLSM.jl]: that sounds good, thank you!!

**Review by juliasloan25 — APPROVED** [review]
Looks great, thank you Kat!

File: experiments/integrated/fluxnet/data_tools.jl (hunk 28 above)
Comment (kmdeck) [reply experiments/integrated/fluxnet/data_tools.jl:66]: good point!  @Espeer5 I will make these changes in a separate PR this week
